pkg/shardddl/optimism/lock.go

```diff
@@ -101,6 +101,24 @@ func (l *Lock) TrySync(callerSource, callerSchema, callerTable string,
 		l.mu.Unlock()
 	}()
 
+	oldSynced := l.synced
+	defer func() {
+		_, remain := l.syncStatus()
+		l.synced = remain == 0
+		if oldSynced != l.synced {
+			if oldSynced {
+				metrics.ReportDDLPending(l.Task, metrics.DDLPendingSynced, metrics.DDLPendingUnSynced)
+			} else {
+				metrics.ReportDDLPending(l.Task, metrics.DDLPendingUnSynced, metrics.DDLPendingSynced)
+			}
+		}
+	}()
+
+	// should not happen
+	if len(ddls) != len(newTIs) || len(newTIs) == 0 {
```

Comment:
will `len(ddls)` be 0 🤔

Reply:
[no](https://github.com/pingcap/dm/blob/cd9e90106c58674c1d864d2a21fa450f41fa1591/syncer/syncer.go#L1869)

Reply:
I think `len(ddls) != len(newTIs)` is enough, it's OK because more robust

Reply:
We use `newTIs[len(newTIs)-1])` below. I'm afraid of panic error. 😂